README.md

```diff
@@ -8,10 +16,122 @@ Apparently *the blue crab* calls this sort of environment home.
-way to passively cache external dependencies in the service so that builds can
-continue even when cut off from the outside world, or during github/crates.io
-outages.
+*Devpi* offers a rich set of features including user-centric indexes,
+package search, passive upstream index mirroring and even a web UI.
+
+Today, *Estuary* only supports the most fundamental registry functions:
+
+- publishing
+- yanking
+- downloading
+
+These features allow us to `cargo install` or use crates from the registry as
+dependencies in other crates.
+
+**Estuary does not yet implement any sort of authentication handling and as
+such is only appropriate for *internal use*.**
+
+
+## Installation
+
+```
+$ cargo install estuary
+```
+
+Estuary depends on being able to run `git` on the command-line.
+
+
+## Usage
+
+### Estuary Server
+
+Estuary relies on environment variables for configuration.
+
+Required:
+
+- `ESTUARY_INDEX_DIR` Directory to store the git repository used to manage the package index.
+- `ESTUARY_CRATE_DIR` Directory path for crate files to be written to.
+- `ESTUARY_API_URL` Public root URL for the Estuary server ex: `http://estuary.example.com` (no trailing slash).
+
+Optional:
+
+- `ESTUARY_HOST` defaults to `0.0.0.0`.
+- `ESTUARY_PORT` defaults to `7878`.
+- `ESTUARY_DL_URL` defaults to `${ESTUARY_API_URL}/api/v1/crates/{crate}/{version}/download`.
+- `ESTUARY_GIT_BIN` defaults to just `git`, expecting it to be in the `PATH`. 
+
+
+When crates are published to Estuary, the `.crate` files are written to
+`${ESTUARY_CRATE_DIR}/{crate}/{version}/{crate}-{version}.crate` which
+aligns with the default value for `ESTUARY_DL_URL`.
+
+
+> If you prefer, you can serve the `.crate` files using a separate web server.
+> For this, you'd set `ESTUARY_DL_URL` to point to that other web server, using
+> the same `{crate}` and `{version}` replacement tokens.
+> 
+> See the [section on `dl`][index format] in the alternate cargo registry docs 
+> for a full list of available tokens cargo can use when building download urls.
+> These other tokens may be useful if you end up moving the crate files into a
+> different directory layout or have URL rewrite rules to deal with.
+
+With your configuration in place, run `estuary` to launch the server.
+
+
+### Configuring Cargo
+
+Estuary exposes the package index git repository at the following URL:
+
+```
+${ESTUARY_API_URL}/git/index
+```
+
+To use Estuary for publishing or installing crates via cargo you need to add
+some configuration. 
+
+For example, if you defined `ESTUARY_API_URL` as `http://estuary.example.com`, you
+would add the following to your `.cargo/config.toml`:
+
+```toml
+[registries]
+estuary = { index = "http://estuary.example.com/git/index" }
+```
+
+With this entry added to your config, the next step is to "authenticate."
+
+```
+$ cargo login --registry estuary
+```
+
+> Note that Estuary currently does with the token to validate access.
```

Comment:
```suggestion
> Note that Estuary currently does nothing with the token to validate access.
```

I'm assuming that was the intent, but there is a word missing in there

Reply:
:smile: :cry: yep. Thanks!